appserver/ejb/ejb-rest-remoting/client/src/main/java/fish/payara/ejb/rest/client/RemoteEJBContext.java

```diff
@@ -0,0 +1,382 @@
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.ejb.rest.client;
+
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+import java.net.URI;
+import java.net.URL;
+import java.security.KeyStore;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+
+import javax.naming.Binding;
+import javax.naming.Context;
+import javax.naming.Name;
+import javax.naming.NameClassPair;
+import javax.naming.NameParser;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.Configuration;
+
+/**
+ * This is the context used for looking up and invoking remote EJBs via
+ * REST in Payara 5.191+.
+ * 
+ * <p>
+ * Note that at the moment only the <code>lookup</code> methods are implemented.
+ * 
+ * <p>
+ * This context supports the following Payara specific properties in its environment:
+ * 
+ * <ul>
+ *     <li>fish.payara.connectTimeout</li>
+ *     <li>fish.payara.executorService</li>
+ *     <li>fish.payara.hostnameVerifier</li>
+ *     <li>fish.payara.keyStore</li>
+ *     <li>fish.payara.readTimeout</li>
+ *     <li>fish.payara.hostnameVerifier</li>
+ *     <li>fish.payara.scheduledExecutorService"</li>
+ *     <li>fish.payara.sslContext</li>
+ *     <li>fish.payara.trustStore</li>
+ *     <li>fish.payara.withConfig</li>
+ * </ul>
+ * 
+ * All properties corresponds to the simularly named settings on the JAX-RS {@link ClientBuilder}.
+ * Times are in microseconds, and values can either be given as Strings, number values or object instances.
+ * 
+ * @author Arjan Tijms
+ * @since Payara 5.191
+ *
+ */
+class RemoteEJBContext implements Context {
```

Comment:
Purposefully not public?

Reply:
Yes, it's a client lib and the classes should be used via `new `InititalContext()` only.